allennlp/modules/token_embedders/pretrained_transformer_mismatched_embedder.py

```diff
@@ -111,15 +119,53 @@ def forward(
         # span_embeddings: (batch_size, num_orig_tokens, max_span_length, embedding_size)
         # span_mask: (batch_size, num_orig_tokens, max_span_length)
         span_embeddings, span_mask = util.batched_span_select(embeddings.contiguous(), offsets)
-        span_mask = span_mask.unsqueeze(-1)
-        span_embeddings *= span_mask  # zero out paddings
 
-        span_embeddings_sum = span_embeddings.sum(2)
-        span_embeddings_len = span_mask.sum(2)
-        # Shape: (batch_size, num_orig_tokens, embedding_size)
-        orig_embeddings = span_embeddings_sum / torch.clamp_min(span_embeddings_len, 1)
+        # int: maximum number of sub-tokens a batch has
+        max_span_length = span_mask.shape[-1]
+
+        # If "sub_token_mode" is set to "first", return the first sub-token embedding
+        if self.sub_token_mode == "first":
+            # Shape: (batch_size, num_orig_tokens, max_span_length)
+            span_mask_bool = torch.zeros(span_mask.shape, dtype=torch.bool)
+
+            # Shape: 1-dimensional tensor
+            max_span_range_indices = torch.arange(max_span_length, dtype=torch.long).view(1, 1, -1)
+
+            # Shape (batch_size, num_orig_tokens, max_span_length)
+            span_mask = max_span_range_indices <= span_mask_bool
```

Comment:
What's going on here? `span_mask_bool` is always 0 at this point.

Reply:
yes, I am constructing the `span_mask` tensor here. Steps are as follows

1. Construct Boolean tensor with the dimension as `(batch_size, num_orig_tokens, max_span_length)`
2. (At line #135) Construct `span_mask` tensor by keeping first tensor from 3rd dimension i.e. `max_span_length`, as Boolean `True` and rest as Boolean `False`. This is done so that when multiplied with `span_embeddings`, it will only select first sub-token embedding and rest all will be zeroed.